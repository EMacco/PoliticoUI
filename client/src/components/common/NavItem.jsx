import React, { Fragment } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const NavItem = ({ name, path }) => {
  return (
    <Fragment>
      <li>
        <Link to={path}>{name}</Link>
      </li>
    </Fragment>
  );
};

NavItem.propTypes = {
  name: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired
};

export default NavItem;
