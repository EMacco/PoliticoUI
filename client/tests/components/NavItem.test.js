import React from 'react';
import NavItem from '@components/common/NavItem';

const props = {
  name: 'Home',
  path: '/'
};

const setup = () => {
  const wrapper = shallow(<NavItem {...props} />);
  return wrapper;
};

describe('<NavItem />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setup();
  });

  it('should render the NavItem component correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper.find('NavItem')).toBeTruthy();
  });
});
