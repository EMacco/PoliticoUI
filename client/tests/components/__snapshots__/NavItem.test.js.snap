// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavItem /> should render the NavItem component correctly 1`] = `
<Fragment>
  <li>
    <Link
      to="/"
    >
      Home
    </Link>
  </li>
</Fragment>
`;
